documentation/docs/migration-guide.md

```diff
@@ -1,10 +1,116 @@
+
+## Migrating to the new node styling API
+
+Before reading this chapter it's recommended to read through [the introduction to the new styling and filtering API](examples/cad-styling).
+
+In the 2.0 of release of Reveal, the filtering and styling API changed drastically - breaking existing
+functionality. Migrating to version 2 will most likely require code changes, but these should often
+be quite limited and lead to a simplification of the code base along with an increase in filtering- and
+styling performance.
+
+In previous versions of Reveal, styling nodes was done by a set of functions for explicitly applying
+styling to a node identified by a tree index (and optionally all its children).
+- Visibility - `hideNodeByTreeIndex()`, `showNodeByTreeIndex()`, `hideAllNodes()` and `showAllNodes()`
+- Color override - `setNodeColorByTreeIndex()`, `setAllNodeColors()`, `resetNodeColorByTreeIndex()` and `resetAllNodeColors()`
+- Ghost nodes - `ghostNodeByTreeIndex()`, `unghostNodeByTreeIndex()`, `ghostAllNodes()` and `unghostAllNodes()`
+- Highlighting - `selectNodeByTreeIndex()`, `deselectNodeByTreeIndex()` and `deselectAllNodes()`
+
+This allowed flexible styling of nodes, but required many individual calls to update each node, causing degraded
+performance. The new API instead requires a set of styles which nodes are assigned to. It's preferable not to have
+too many styles - both in terms of performance and to avoid a cluttered view for the end user.
+
+The previous styling options maps quite directly to the new API. Visibility is controlled by `NodeApperance.visible`,
+color override by `NodeApperance.color` and ghost mode by `NodeApperance.renderGhosted`. However, there is no
+direct mapping between what was previously called 'selection' or 'highlighting', this is now a combination of
+`NodeApperance.color`, `NodeApperance.renderInFront` and `NodeApperance.outlineColor`. This means that the new API
+introduces additional flexibility - e.g. can a object have an outline, but not be rendered in front of other objects. For
+convinience, the new API comes with a sef of default styles available through [`DefaultNodeAppearance`](examples/cad-styling).
+
+A commonly used functionality is to apply a default style to all nodes and to style a few selected nodes with a different
+style to make them stand out. Before, this could be achieved using a `ghostAllNodes`, `hideAllNodes` and `setAllNodeColors` to
+apply the default styling, and a combination of the different per-node styling functions mentioned above to style the 
+nodes that are to stand out. In Reveal 2, this is replaced by `Cognite3DModel.setDefaultNodeAppearance` which is a style that will be applied to any node that is not styled by another styled set.
```

Comment:
```suggestion
There are some important differences between version 1 and version 2 of `@cognite/reveal` which might require changes in the build configuration and code. The most important changes are:

- The node styling/filtering API has been reworked and will require changes. See below for more information.
- [ThreeJS](https://www.npmjs.com/package/three) and [@types/three](https://www.npmjs.com/package/@types/three) are included in the package, so it's no longer necessary to include them in the application you are using Reveal from if you are not using ThreeJS directly.

## Migrating to the new node styling API

Before reading this chapter it's recommended to read through [the introduction to the new styling and filtering API](examples/cad-styling).

In the 2.0 release of Reveal, the filtering and styling API changed drastically - breaking existing
functionality. Migrating to version 2 will most likely require code changes, but these should often
be quite limited and lead to a simplification of your codebase along with an increase in filtering- and
styling performance.

In previous versions of Reveal, nodes styling was done by a set of functions for explicit applying of
styling to a node identified by a tree index (and optionally all its children).

- Visibility - `hideNodeByTreeIndex()`, `showNodeByTreeIndex()`, `hideAllNodes()` and `showAllNodes()`
- Color override - `setNodeColorByTreeIndex()`, `setAllNodeColors()`, `resetNodeColorByTreeIndex()` and `resetAllNodeColors()`
- Ghost nodes - `ghostNodeByTreeIndex()`, `unghostNodeByTreeIndex()`, `ghostAllNodes()` and `unghostAllNodes()`
- Highlighting - `selectNodeByTreeIndex()`, `deselectNodeByTreeIndex()` and `deselectAllNodes()`

This allowed flexible styling of nodes but required many individual calls to update each node, causing degraded
performance. 
The new API instead requires a set of styles that are assigned to nodes. 
It's preferable not to have too many styles - both in terms of performance and to avoid a cluttered view for the end user.

The previous styling options maps quite directly to the new API. Visibility is controlled by `NodeApperance.visible`,
color override by `NodeApperance.color` and ghost mode by `NodeApperance.renderGhosted`. However, there is no
direct mapping between what was previously called 'selection' or 'highlighting', this is now a combination of
`NodeApperance.color`, `NodeApperance.renderInFront` and `NodeApperance.outlineColor`. This means that the new API
introduces additional flexibility - e.g. an object can have an outline, but not be rendered in front of other objects. For
convenience, the new API comes with a set of default styles available through [`DefaultNodeAppearance`](examples/cad-styling).

A commonly used functionality is to apply a default style to all nodes and to style a few selected nodes with a different
style to make them stand out. Before, this could be achieved using a `ghostAllNodes`, `hideAllNodes` and `setAllNodeColors` to
apply the default styling, and a combination of the different per-node styling functions mentioned above to style the 
nodes that are to stand out. In Reveal 2, this is replaced by `Cognite3DModel.setDefaultNodeAppearance` which is a style that will be applied to any node that is not styled by another styled set.
```

Diff looks ugly, sorry about that. See actual changes:

![image](https://user-images.githubusercontent.com/4989157/120343624-2a76b500-c2f9-11eb-9676-c734861705f5.png)

Tried to proofread here a bit. @larsmoa great work on the migration guide 💪